File: imagepipeline-base/src/main/java/com/facebook/cache/common/HasDebugData.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.cache.common;

import com.facebook.infer.annotation.Nullsafe;
import javax.annotation.Nullable;

@Nullsafe(Nullsafe.Mode.STRICT)
public interface HasDebugData {

  /** Return a string describing this object and to be used for debugging or logging */
  @Nullable
  String getDebugData();
}

File: imagepipeline-base/src/main/java/com/facebook/cache/common/HasDebugData.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.cache.common

interface HasDebugData {
  /** Return a string describing this object and to be used for debugging or logging */
  val debugData: String?
}
